### PR TITLE
feat(node_parser): add SlideNodeParser integration

### PR DESCRIPTION
**Pull request by AnishJoshi13**
# Description

Adds a brand‐new SlideNodeParser integration to LlamaIndex. This implements the SLIDE algorithm (Sliding Localized Information for Document Extraction) from the paper [SLIDE: Sliding Localized Information for Document Extraction](https://arxiv.org/abs/2503.17952) which:

- Splits documents into sentences using a punkt tokenizer

- Greedily groups sentences into token-bounded “base” chunks

- Uses a sliding window of neighboring chunks to generate a short, LLM-generated context for each chunk

- Attaches that localized context as node metadata to improve downstream retrieval, QA, and knowledge-graph extraction

**Motivation**: Improves GraphRAG performance on multilingual and low-resource documents by preserving critical local context that would otherwise be lost when documents exceed an LLM’s context window.

This PR includes:

- The full parser implementation (```base.py```) with robust Pydantic validation, type hinting, edge-case handling, and model validators

- edge and integrations tests (```test_node_parser_slide.py```) covering empty docs, per-chunk LLM calls, and other behaviors

- ```README.md```, API reference (```slide.md```). Will add example notebook (```slide_parser.ipynb```) once integration is pushed to PyPI.

- A ```Makefile``` and a ```pyproject.toml```  to match upstream CI conventions. Will add ```uv.lock``` upon request.

No external dependencies beyond ```llama-index-core```.

## New Package?

Did I fill in the `tool.llamahub` section in the `pyproject.toml` and provide a detailed README.md for my new integration or package?

- [x] Yes
- [ ] No

## Version Bump?

Did I bump the version in the `pyproject.toml` file of the package I am updating? (Except for the `llama-index-core` package)

- [ ] Yes
- [x] No (new integration package; initial version is set in ```pyproject.toml```

## Type of Change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## How Has This Been Tested?

Your pull-request will likely not be merged unless it is covered by some form of impactful unit testing.

- [x] I added new unit tests to cover this change
- [x] I believe this change is already covered by existing unit tests

## Suggested Checklist:

- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation (```README.md```, API reference)
- [ ] I have added Google Colab support for the newly added notebooks.
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] I ran `make format; make lint` to appease the lint gods

### CODE REVIEW

File: llama-index-integrations/node_parser/llama-index-node-parser-slide/llama_index/node_parser/slide/base.py [hunk 1]
```diff
@@ -0,0 +1,226 @@
...
+            # 5) Warn if context_window not provided
+            warnings.warn(
+                "The LLM does not expose `metadata.context_window`. "
+                "SLIDE cannot validate token usage, which may lead to truncation or generation failures.",
+                stacklevel=2
+            )
+
+        return self
+
+    def _parse_nodes(
```
Comment (logan-markewich) [inline llama-index-integrations/node_parser/llama-index-node-parser-slide/llama_index/node_parser/slide/base.py:126]: Since we are using an llm, it might be helpful to also implement `async def _aparse_nodes()` -- that way we can use `llm.achat(..)` and possibly even run llm calls concurrently using `run_jobs` from `llama_index.core.async_utils` 👀

**Comment by logan-markewich** [comment]
The paper doesn't discuss it, but this feels like a more specific version of contextual retrieval from anthropic (which we also support as a metadata extractor)
https://docs.llamaindex.ai/en/stable/examples/metadata_extraction/DocumentContextExtractor/

I wonder how it compares in terms of retrieval performance

File: llama-index-integrations/node_parser/llama-index-node-parser-slide/llama_index/node_parser/slide/base.py (hunk 1 above)
Comment (logan-markewich) [reply llama-index-integrations/node_parser/llama-index-node-parser-slide/llama_index/node_parser/slide/base.py:126]: This would be especially helpful for users who want to run this in production, since async is very important

Comment (AnishJoshi13) [reply llama-index-integrations/node_parser/llama-index-node-parser-slide/llama_index/node_parser/slide/base.py:126]: Yes, thought about it previously. Definitely sounds like a good idea. Was busy for 2 days, I'm implementing this now. ```llama_index.core.async_utils``` is something which I missed before, this will surely help

**Comment by AnishJoshi13** [comment]
## Summary of Changes

### base.py
- Added `llm_workers: int` Pydantic field (default=1) to control async parallelism.
- Added `stacklevel=2` runtime `warnings.warn(...)` in `_parse_nodes` to explicitly notify users if they have set `llm_workers > 1` in sync mode — advising them to use the async `aget_nodes_from_documents(...)` method instead.
- Introduced `abuild_localised_splits(...)` to batch LLM calls via `run_jobs`.
- Implemented async methods:
  - `async def _aparse_nodes(...)`
  - `async def abuild_slide_nodes_from_documents(...)`
- Updated async logic to call `run_jobs(jobs, workers=self.llm_workers, ...)` for throttled parallel LLM calls.

### test_node_parser_slide.py
- Synchronous parsing (`get_nodes_from_documents`) under various chunk_size, window_size, and llm_workers.
- Asynchronous parsing (`aget_nodes_from_documents`) under the same parameter sets.
- Parallelism of async LLM calls (`achat`) via run_jobs, ensuring overlap when llm_workers > 1.

### README.md
- Documented the new `llm_workers` parameter and its default of `1`.
- Updated usage examples to show both sync and async entrypoints:
    ```python
    # Sync
    nodes = parser.get_nodes_from_documents(docs)

    # Async
    nodes = await parser.aget_nodes_from_documents(docs)
    ```
- Clarified how to configure `chunk_size`, `window_size`, and `llm_workers` for optimal parallelism.

**Review by logan-markewich — APPROVED** [review]
LGTM!

**Comment by AnishJoshi13** [comment]
Hey I tried to rebase merge them, looks like there are still some workflows that require your approval
